src/app/services/theme/theme.service.ts

```diff
@@ -0,0 +1,149 @@
+import { DOCUMENT } from "@angular/common";
+import { Inject, Injectable } from "@angular/core";
+import { hsl, HSLColor } from "d3-color";
+
+const themeColors = [
+  "highlight",
+  "primary",
+  "secondary",
+  "success",
+  "info",
+  "warning",
+  "danger",
+  "light",
+  "dark",
+] as const;
+export type ThemeColor = typeof themeColors[number];
+
+const themeVariants = [
+  "",
+  "-lighter",
+  "-lightest",
+  "-darker",
+  "-darkest",
+] as const;
+export type ThemeVariant = typeof themeVariants[number];
+
+/** Configuration theme settings */
+export type BawTheme = { [color in ThemeColor]?: string };
+
+@Injectable({
+  providedIn: "root",
+})
+export class ThemeService {
+  /** List of colour options in theme */
+  public themeColors = themeColors;
+  /** List of variants for each colour option in theme */
+  public themeVariants = themeVariants;
+
+  /** Tracks config changes to theme */
+  private theme: BawTheme;
+  /** Stored styles of the website */
+  private style: CSSStyleDeclaration;
+
+  public constructor(@Inject(DOCUMENT) private document: Document) {
+    // Read the root style of the website
+    this.style = this.document.documentElement.style;
```

Comment:
hmm, it would not have been my first instinct to store the style map over the dom element reference.

Is there a reason?

Reply:
I never use the dom element, so everywhere in my code was being written as `root.style`. Potentially that provides more semantic information, but i found it repetitive.